src/core/lib/event_engine/posix_engine/ev_epoll1_linux.cc

```diff
@@ -487,24 +523,25 @@ void Epoll1EventHandle::SetWritable() { write_closure_->SetReady(); }
 
 void Epoll1EventHandle::SetHasError() { error_closure_->SetReady(); }
 
-absl::Status Epoll1Poller::Work(grpc_core::Timestamp deadline,
-                                std::vector<EventHandle*>& pending_events) {
+Poller::WorkResult Epoll1Poller::Work(EventEngine::Duration timeout) {
+  Poller::Events pending_events;
   if (g_epoll_set_.cursor == g_epoll_set_.num_events) {
-    auto status = DoEpollWait(deadline);
-    if (!status.ok()) {
-      return status;
+    if (DoEpollWait(timeout) == 0) {
+      return Poller::DeadlineExceeded{};
     }
   }
   {
     absl::MutexLock lock(&mu_);
     // If was_kicked_ is true, collect all pending events in this iteration.
-    auto status = ProcessEpollEvents(
-        was_kicked_ ? INT_MAX : MAX_EPOLL_EVENTS_HANDLED_PER_ITERATION,
-        pending_events);
-    if (!status.ok()) {
+    if (ProcessEpollEvents(
+            was_kicked_ ? INT_MAX : MAX_EPOLL_EVENTS_HANDLED_PER_ITERATION,
+            pending_events)) {
       was_kicked_ = false;
     }
-    return status;
+    if (pending_events.empty()) {
+      return Poller::Kicked{};
+    }
+    return pending_events;
```

Comment:
nit: it seems non-intuitive that the poller could be kicked, but Work returns a result that is not Poller::Kicked(). I think it's worth documenting.

Reply:
Added a comment to Epoll1Poller::Work